### PR TITLE
Binary search tree in Dart language

### LINKED ISSUE #2199: Binary Search Trees
- [x] C
- [x] C++
- [x] C#
- [ ] CoffeeScript
- [ ] Dart
- [ ] Go
- [x] Java
- [x] JS
- [ ] Kotlin
- [ ] Php
- [x] Python
- [x] Readme
- [x] Ruby
- [ ] Rust
- [ ] TypeScript

### PR DESCRIPTION
**Pull request by keshakaneria**
Fixes #2199 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Languages Used:
- Dart

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

**Comment by keshakaneria** [comment]
@jainaman224 irrelevant files deleted and binary search tree file added with dynamic values

**Comment by keshakaneria** [comment]
https://github.com/jainaman224/Algo_Ds_Notes/pull/2429#pullrequestreview-375007075
@asha15 spaces after which brakets? and indentation in which lines?

**Comment by keshakaneria** [comment]
> correct the indentation
> remove the space after bracket
> squash the commit and push it

@asha15 spaces after which brackets? and indentation in which lines?

File: Binary_Search_Trees/BinarySearchTree.dart [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+
+	Node(int val){
+		this.value = val;
+		this.leftNode = null;
+		this.rightNode = null;
+	}
+	
+	Node get left{
+		return this.leftNode;
+	}
```
Comment (asha15) [inline Binary_Search_Trees/BinarySearchTree.dart]: keep four spaces instead of tab

**Comment by keshakaneria** [comment]
Kindly review this PR @asha15

File: Binary_Search_Trees/BinarySearchTree.dart [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+}
+
+postOrderTraversal(Node x){
+    if(x == null) return;
+    postOrderTraversal(x.left);
+    print(x.val);
+    postOrderTraversal(x.right);
+}
+
+insertElement(Node x,int val){
```
Comment (asha15) [inline Binary_Search_Trees/BinarySearchTree.dart:62]: keep a space after comma

**Comment by somya-kapoor** [comment]
Are you working on it?

**Comment by keshakaneria** [comment]
@somya-kapoor Yes i am, the code is almost ready just need to squash and submit again after requested change is done.

**Comment by keshakaneria** [comment]
@asha15 Can you help me out to squash commits because i am not using git to push and pull commits, doing directly from branch itself. So how can I squash from github interface?

**Comment by asha15** [comment]
we only have the permission to quash it using interface.

I kindly request you to refer some tutorials to do squash using git bash. if you find it difficult, please reach me on slack

**Comment by keshakaneria** [comment]
@asha15 I have squashed the commits and created a new PR with new branch so that no future confusion is there for me to work. Kindly verify that new PR and evaluate me according to difficulty of Issue.

**Comment by asha15** [comment]
next time please create a PR from Git Bash. Grab this opportunity to learn

**Comment by keshakaneria** [comment]
@asha15 Sure Thank you so much! can you just add label this so that i can be evaluated for GSSoC with difficulty level of issue

**Pull request closed without merging**